code/datums/mutations/touch.dm

```diff
@@ -244,26 +273,65 @@
 		if(!(mendicant_transfer_limb in mendicant_organic_limbs))
 			mendicant_transfer_limb = pick(mendicant_organic_limbs)
 
-		// Transfer at most 35 damage by default.
+		// Transfer at most 35 damage, by default.
 		var/brute_damage = min(affected_limb.brute_dam, 35 * heal_multiplier)
 		// no double dipping
 		var/burn_damage = min(affected_limb.burn_dam, (35 * heal_multiplier) - brute_damage)
 		if((brute_damage || burn_damage) && total_damage_healed < (35 * heal_multiplier))
-			total_damage_healed = brute_damage + burn_damage
+			total_damage_healed += brute_damage + burn_damage
 			. = TRUE
+			var/brute_taken = brute_damage * pain_multiplier
+			var/burn_taken = burn_damage * pain_multiplier
 			// Heal!
-			affected_limb.heal_damage(brute_damage * heal_multiplier, burn_damage * heal_multiplier, required_bodytype = BODYTYPE_ORGANIC)
+			affected_limb.heal_damage(brute_damage, burn_damage, required_bodytype = BODYTYPE_ORGANIC)
 			// Hurt!
-			mendicant_transfer_limb.receive_damage(brute_damage * pain_multiplier, burn_damage * pain_multiplier, forced = TRUE, wound_bonus = CANT_WOUND)
+			mendicant_transfer_limb.receive_damage(brute_taken, burn_taken, forced = TRUE, wound_bonus = CANT_WOUND)
 
 		// Force light wounds onto you.
 		for(var/datum/wound/iter_wound as anything in affected_limb.wounds)
-			if(iter_wound.severity > WOUND_SEVERITY_MODERATE)
-				continue
+			switch(iter_wound.severity)
+				if(WOUND_SEVERITY_SEVERE) // half and half
+					if(prob(50 * heal_multiplier))
+						continue
+				if(WOUND_SEVERITY_CRITICAL)
+					if(heal_multiplier < 1.5) // need buffs to transfer crit wounds
+						continue
 			. = TRUE
 			iter_wound.remove_wound()
 			iter_wound.apply_wound(mendicant_transfer_limb)
 
+	// 10% base
+	var/max_blood_transfer = (BLOOD_VOLUME_NORMAL * 0.10) * heal_multiplier
+	if(!HAS_TRAIT(mendicant, TRAIT_NOBLOOD))
+		if(hurtguy.blood_volume < BLOOD_VOLUME_NORMAL)
+			var/max_blood_to_hurtguy = min(mendicant.blood_volume, BLOOD_VOLUME_NORMAL - hurtguy.blood_volume)
+			var/blood_to_hurtguy = min(max_blood_transfer, max_blood_to_hurtguy)
+			if(blood_to_hurtguy)
+				// We ignore incompatibility here.
+				mendicant.transfer_blood_to(hurtguy, blood_to_hurtguy, forced = TRUE, ignore_incompatibility = TRUE)
+				to_chat(mendicant, span_notice("Your veins (and brain) feel a bit lighter."))
+				. = TRUE
+				// Because we do our own spin on it!
+				if(hurtguy.get_blood_compatibility(mendicant) == FALSE)
+					hurtguy.adjustToxLoss((blood_to_hurtguy * 0.1) * pain_multiplier) // 1 dmg per 10 blood
+					to_chat(hurtguy, span_notice("Your veins feel thicker, but they itch a bit."))
+				else
+					to_chat(hurtguy, span_notice("Your veins feel thicker!"))
+
+		if(hurtguy.blood_volume > BLOOD_VOLUME_MAXIMUM)
```

Comment:
here

Reply:
you skipped this. the `volume_normal` block can be safely returned, there's no case where they're both under and over. 
this could be early returned to make the proceeding block unindented